src/source/rtl_text_plugin.js

```diff
@@ -1,50 +1,73 @@
 // @flow
 
 import {Event, Evented} from '../util/evented';
+import {getArrayBuffer} from '../util/ajax';
 import browser from '../util/browser';
+import window from '../util/window';
 
 const status = {
-    unavailable: 'unavailable',
-    loading: 'loading',
+    unavailable: 'unavailable', // Not loaded
+    available: 'available', // Host url specified, but we havent started loading yet
+    loading: 'loading', // request in-flight
+    downloaded: 'downloaded', //plugin loaded and cached on main-thread and pluginBlobURL for worker is generated
```

Comment:
What is the difference between `downloaded` and `loaded`, and why would we want to expose that distinction?

Reply:
Removed, I've changed the architecture as described in the main PR body.